bin/cpu_monitor.py

```diff
@@ -512,14 +529,21 @@ def publish_stats(self):
             update_status_stale(self._temp_stat, self._last_temp_time)
             update_status_stale(self._usage_stat, self._last_usage_time)
 
-            msg = DiagnosticArray()
-            msg.header.stamp = rospy.get_rostime()
-            msg.status.append(self._temp_stat)
-            msg.status.append(self._usage_stat)
-
-            if rospy.get_time() - self._last_publish_time > 0.5:
-                self._diag_pub.publish(msg)
-                self._last_publish_time = rospy.get_time()
+            # Convert from ROS diagnostics to mbot_diagnostics for publishing.
+            self._temp_diagnostic.set_status(
```

Comment:
It looks like this is creating a D1.0 status message, updating the properties of that message, then when publishing, copying those properties back to the 2.0 GenericDiagnostic. Why not update just update the GenericDiagnostic component directly and get rid of the D1.0 intermediary?

This comment applies to all the files in this PR.

Reply:
Correct, instead of making pretty heavy-handed updates to the internals of many monitor functions, I thought it might be reasonable to just change the interface to the rest of the system - internal work in 1.0, publishing via 2.0. A bit of a quick-and-dirty compromise. This being forked code that does not necessarily need to be uniform with our codebase definitely played a critical role in that decision. What are your thoughts on that, before I do any re-writing?

Reply:
Yeah it is forked so if we were trying to keep it current with upstream commits I could see the motivation for minimizing changes. I think we've made enough changes to it that we probably won't be pulling in updates. That said, what you have works so it's fine to keep as-is, just an opportunity to simplify.

Reply:
Yeah, agreed that it's a partially moot point since we're not trying to pull in/keep up-to-date with upstream. It is kinda nice to see a minimal list of changes from upstream (assuming that's stable) if we're ever trying to track down tricky bugs in the future that we may have introduced. But overall mainly just trying to keep the number of changes we make to known/assumed stable code minimal, especially if additional changes are not mission-critical.

Anyway, thanks for discussing this and marking resolved.